be/src/runtime/thread_mem_tracker_mgr.h

```diff
@@ -227,21 +253,36 @@ inline void ThreadMemTrackerMgr::cache_consume(int64_t size) {
             _untracked_mem += _untracked_mems[_tracker_id];
             _untracked_mems[_tracker_id] = 0;
         }
-        noncache_consume();
+        noncache_consume(_untracked_mem);
+        _untracked_mem = 0;
         start_thread_mem_tracker = true;
     }
 }
 
-inline void ThreadMemTrackerMgr::noncache_consume() {
-    DCHECK(_mem_trackers[_tracker_id]) << ", label: " << _mem_tracker_labels[_tracker_id];
-    Status st = mem_tracker()->try_consume(_untracked_mem);
+inline void ThreadMemTrackerMgr::noncache_consume(int64_t size) {
+    Status st = mem_tracker()->try_consume(size);
     if (!st) {
         // The memory has been allocated, so when TryConsume fails, need to continue to complete
         // the consume to ensure the accuracy of the statistics.
-        mem_tracker()->consume(_untracked_mem);
-        exceeded(_untracked_mem, st);
+        mem_tracker()->consume(size);
+        exceeded(size, st);
     }
-    _untracked_mem = 0;
+}
+
+inline void ThreadMemTrackerMgr::add_tracker(const std::shared_ptr<MemTracker>& mem_tracker) {
+    DCHECK(_mem_trackers.find(mem_tracker->id()) == _mem_trackers.end()) << print_debug_string();
+    if (_mem_trackers.find(mem_tracker->id()) == _mem_trackers.end()) {
+        _mem_trackers[mem_tracker->id()] = mem_tracker;
+        DCHECK(_mem_trackers[mem_tracker->id()]) << print_debug_string();
```

Comment:
If there is no `DCHECK(_mem_trackers.find(mem_tracker->id()) == _mem_trackers.end())`, the add tracker may be repeated (currently prohibited, for the reasons above). This will, 1. waste time 2. `_untracked_mems[mem_tracker->id()] = 0` will cause the memory track to be lost.

If there is no `DCHECK(_mem_trackers[mem_tracker->id()])` , it may cause empty trackers to be added, which will cause exceptions to consume trackers later.
The reason why it is placed after `_mem_trackers[mem_tracker->id()] = mem_tracker;` is because the parameter is a reference to shared_ptr to avoid the tracker being destructed after DCHECK.